pints/_mcmc/_slice_covariance_matching.py

```diff
@@ -0,0 +1,354 @@
+        """
+        Sets standard deviation of initial crumb.
+        """
+        sigma_c = float(sigma_c)
+        if sigma_c < 0:
+            raise ValueError('Inital crumb standard deviation'
+                             'must be positive.')
+        self._sigma_c = sigma_c
+
+    def set_theta(self, theta):
+        """
+        Sets parameter to control how fast the precision of the
+        proposal distribution increases.
+        """
+        self._theta = float(theta)
+
+    def sigma_c(self):
+        """
+        Returns standard deviation of initial crumb.
+        """
+        return self._sigma_c
+
+    def tell(self, reply):
+        """ See :meth:`pints.SingleChainMCMC.tell()`. """
+
+        # Check ask/tell pattern
+        if not self._ready_for_tell:
+            raise RuntimeError('Tell called before proposal was set.')
+        self._ready_for_tell = False
+
+        # Unpack reply
+        fx, grad = reply
+
+        # Check reply, copy gradient
+        fx = float(fx)
+        grad = pints.vector(grad)
+
+        # If this is the log_pdf of a new point, save the value and use it
+        # to check ``f(x_1) >= y``
+        if self._sent_proposal:
+            self._proposed_pdf = fx
+            self._sent_proposal = False
+
+        # Very first call
+        if self._current is None:
+            # Check first point is somewhere sensible
+            if not np.isfinite(fx):
+                raise ValueError(
+                    'Initial point for MCMC must have finite logpdf.')
+
+            # Set current sample, log pdf of current sample and initialise
+            # proposed sample for next iteration
+            self._current = np.array(self._x0, copy=True)
+            self._proposed = np.array(self._current, copy=True)
+            self._current_log_pdf = fx
+            self._M = fx
+
+            # Define Cholesky upper triangles: F_k for W_k and R_k for Lambda_k
+            self._R = self._sigma_c ** (-1) * np.identity(self._n_parameters)
+            self._F = self._sigma_c ** (-1) * np.identity(self._n_parameters)
+
+            # Sample height of the slice log_y for next iteration
+            e = np.random.exponential(1)
+            self._current_log_y = self._M - e
+
+            # Return first point in chain, which is x0
+            return np.array(self._current, copy=True)
+
+        # Acceptance check
+        if self._proposed_pdf >= self._current_log_y:
+            # The accepted sample becomes the new current sample
+            self._current = np.array(self._proposed, copy=True)
+
+            # Update estimate of mode density
+            self._M = fx
+
+            # Sample new log_y used to define the next slice
+            e = np.random.exponential(1)
+            self._current_log_y = self._M - e
+
+            # Reset parameters
+            self._c_bar_star = np.zeros(self._n_parameters)
+            self._F = self._sigma_c ** (-1) * np.identity(self._n_parameters)
```

Comment:
Look ok although can avoid recreating zero vectors and identity matrices each time